ckanext/restricted/templates/restricted/restricted_string.html

```diff
@@ -0,0 +1,50 @@
+{% resource 'ckanext-restricted/restricted_ui.js' %}
+
+{% set organization_link = h.link_to(pkg.organization.title, h.url_for(controller="organization", action="read", id=pkg.organization.name)) %}
+{% set private_resource_string = _("Access restricted to members of {organisation_name}").format(organisation_name=organization_link) %}
+
+{% if "restricted" not in res or pkg["private"] %}
+
+  {{private_resource_string|safe }}
+
+{% else %}
+
+  {% set restricted = h.restricted_json_loads(res["restricted"]) %}
+
+  {% if restricted["level"] == "public" %}
+
+      {{_('Public resource')}}
+
+  {% else %}
+
+    {% set popup_link = "<a href='#' data-module='restricted_popup' data-module-title='{}' data-module-content='{}'>{}</a>" %}
+    {% set org_string = False %}
+    {% set usr_string = False %}
+
+    {% if restricted.get("allowed_organizations") %}
+      {% set organizations = restricted.get("allowed_organizations").split(",") %}
+      {% set title = _('Access granted to organizations') %}
+      {% set content = "<br />".join(organizations + [pkg.organization.name] ) %}
+      {% set org_string = popup_link.format(title, content, _('organizations')) %}
+    {% endif %}
+
+    {% if restricted.get("allowed_users") %}
+      {% set users = restricted.get("allowed_users").split(",") %}
+      {% set title = _('Access granted to users') %}
+      {% set content = "<br />".join(users) %}
+      {% set usr_string = popup_link.format(title, content, _('users')) %}
+    {% endif %}
```

Comment:
I don't think I can improve the logic by refactoring myself: I spent a while devising this logic as the simplest it can be in my head.  I do think some of the variable names are misleading though, so I will tweak that.